posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd

```diff
@@ -0,0 +1,359 @@
+set.seed(123)
+ex_adsl <- data.frame(
+  ARM = sample(c("Placebo", "Drug X", "Combination"), 100, replace = TRUE),
+  BMRKR2 = sample(c("Low", "Medium", "High"), 100, replace = TRUE),
+  RACE = sample(c("Asian", "Black", "White"), 100, replace = TRUE),
+  SEX = sample(c("Male", "Female"), 100, replace = TRUE),
+  BMRKR1 = rnorm(100, mean = 5, sd = 2)
+)
+
+# Define a function to calculate the mean of a biomarker
+biomarker_ave <- function(x, ...) {
+  val <- if (length(x) > 0) round(mean(x), 2) else "no data"
+  in_rows("Biomarker 1 (mean)" = rcell(val))
+}
+
+# Create the table
+table <- basic_table() |>
+  split_cols_by("ARM") |>
+  split_cols_by("BMRKR2") |>
+  add_colcounts() |>
+  split_rows_by("RACE", split_fun = trim_levels_in_group("SEX")) |>
+  split_rows_by("SEX") |>
+  summarize_row_groups() |>
+  analyze("BMRKR1", biomarker_ave) |>
+  build_table(ex_adsl)
+
+# Print the table
+print(table)
+```
+
+**Results**
+
+![](images/rtables-example.png)
+
+**You can learn more about rtables:**
+
+-   [A Not So Short Introduction to rtables](https://waddella.github.io/RStudioTableContest2020/A_Not_So_Short_Introduction_to_rtables.html) 
+
+-   [Introduction to {rtables}](https://cran.r-project.org/web/packages/rtables/vignettes/introduction.html)
+
+-   [GitHub - insightsengineering/rtables: Reporting tables with R](https://github.com/insightsengineering/rtables)  
+
+## **{admiral}: Simplifying ADaM Dataset Creation**
+
+![](images/admiral.png){width="260"}
+
+Next let’s take a look at the [**{admiral}**](https://pharmaverse.github.io/admiral/) package, developed through a collaboration between [Roche](https://www.roche.com/) and [GSK](https://www.gsk.com/en-gb/). It provides a toolbox of reusable functions and utilities with [**dplyr**](https://dplyr.tidyverse.org/)**-like syntax** to prepare **CDISC ADaM (Analysis Data Model)** **datasets**. It serves as a valuable resource for statistical programmers looking to build ADaMs according to varying analysis needs while ensuring traceability and compliance with **FDA standards**.
+
+### **Features of {admiral}**
+
+Let’s take a look at some features of the package: 
+
+-   [{admiral}](https://github.com/pharmaverse/admiral/) offers **modular functions** for adding variables and records to datasets in a step-by-step manner. This makes it easy to adjust code by adding, removing, or modifying derivations.
+
+-   It provides **template R scripts as a starting point for creating different ADaM datasets.** These templates can be generated using the use_ad_template() function.
+
+-   The package offers a **flexible and efficient way to generate ADaM datasets** from source data, ensuring compliance with CDISC standards and traceability of the data transformation process.
+
+-   **By providing a consistent set of functions and utilities** for ADaM dataset creation, admiral helps to standardize the reporting process across different clinical trials and pharmaceutical companies.
```

Comment:
"admiral" is sometimes within curly brackets and sometimes not, and sometimes it's using a link and sometimes not - i would always use curly brackets and a link.